python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1057,6 +1060,62 @@ def get_asset_backfill_iteration_materialized_partitions(
     yield updated_materialized_subset
 
 
+def get_run_success_but_not_materialized_subset(
+    backfill_id: str,
+    asset_backfill_data: AssetBackfillData,
+    asset_graph: ExternalAssetGraph,
+    instance_queryer: CachingInstanceQueryer,
+) -> Iterable[Optional[AssetGraphSubset]]:
+    run_success_but_not_materialized_partitions = AssetGraphSubset(asset_graph)
+
+    successful_run_ids = [
+        record.run_id
+        for record in instance_queryer.instance.get_event_records(
+            EventRecordsFilter(
+                event_type=DagsterEventType.RUN_SUCCESS,
+                after_cursor=asset_backfill_data.latest_storage_id,
+            ),
+            force_use_index_connection=True,
+        )
+    ]
+
+    yield None
+
+    backfill_run_ids = []
+
+    for run_id in successful_run_ids:
```

Comment:
there's opportunity to consolidate this with the logic in https://github.com/dagster-io/dagster/blob/57df2a919ce3b428cd03665c83129ae8febafa19/python_modules/dagster/dagster/_core/execution/asset_backfill.py#L1292-L1331